### ISSUE TITLE
global-require: allow allowlist for require paths

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#11300]
**What rule do you want to change?**

global-require

**Does this change cause the rule to produce more or fewer warnings?**

Fewer.

It's common in projects using bundlers such as Webpack to inline require resources such as images.
`global-require` will always error, even though those lines are valid specifically for those resource types.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option to allow `global-require` to take in an allowlist of paths that shouldn't produce complaints (#9028).

For example, to always allow requiring JPG images inline:

```json
{
    "global-require": [
        true,
        {
            "allowlist": [
                "\\.jpg$"
            ]
        }
    ]
}
```

**Please provide some example code that this change will affect:**

```js
const RendersImage = () => {
    const imageData = require("./images/image.jpg");

    return (
        <div style={{ backgroundImage: `url(${imageData})` })} />
    );
};
```

**What does the rule currently do for this code?**

Complains that the `require("./images/image.jpg");` call is not global.

**What will the rule do after it's changed?**

No longer complain on that line, but still for any other non-glboal requires not allowlisted.

**Are you willing to submit a pull request to implement this change?**

Yes! 🙌

**Comment by platinumazure** [comment]
Sounds reasonable to me. I'll champion this.

**Comment by kaicataldo** [comment]
@platinumazure Are you still championing this?

**Comment by JoshuaKGoldberg** [comment]
Updated to say _allow_list instead of _white_list per https://eslint.org/blog/2020/08/making-eslint-more-inclusive.

@platinumazure sorry to re-ping you on this, but since it's been a while - is there anything I can do to help push this forward?

**Comment by JoshuaKGoldberg** [comment]
@kaicataldo, redirecting to you 😄 is there anything that can be done here.

**Comment by mdjermanovic** [comment]
@JoshuaKGoldberg

This enhancement was close to getting support from the team, but the rule in question has been deprecated since, so we won't be able to accept the enhancement now. 

`global-require` and other Node.js and CommonJS rules were deprecated in `v7.0.0` and moved to [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node).

You can find more details here:

https://eslint.org/docs/user-guide/migrating-to-7.0.0#nodejscommonjs-rules-have-been-deprecated

Per our [policy about deprecated rules](https://eslint.org/docs/user-guide/rule-deprecation), we are no longer making any updates or bug fixes to the versions of Node.js and CommonJS rules we now have in the core, so it's highly recommended to switch to their `eslint-plugin-node` replacements.

This also means that any enhancement requests for these rules should be submitted to the `eslint-plugin-node` repository:

https://github.com/mysticatea/eslint-plugin-node

This is the replacement rule:

https://github.com/mysticatea/eslint-plugin-node/blob/v11.1.0/docs/rules/global-require.md

**Comment by JoshuaKGoldberg** [comment]
Oh, perfect, that works for me. Thanks!
